pkg/appgateserver/cmd/cmd.go

```diff
@@ -0,0 +1,199 @@
+	cmd := &cobra.Command{
+		Use:   "appgate-server",
+		Short: "Starts the AppGate server",
+		Long: `Starts the AppGate server that will listen for incoming relays requests and will handle
+the interaction with the chain, sessions and suppliers in order to receive the correct
+response for the request.
+
+If the server is started with a defined --self-signing flag, it will behave
+as an application and sign any incoming requests with the private key associated with
+the --signing-key-name flag. If however, this flag is not provided, the server will
+behave as a gateway and will sign relays on behalf of any application sending it relays provided
+that the address recieved in the query parameters of a request has been delegated to by the
+gateway, this is so that it can sign relays using the ring of the application with the
+key associated with the --signing-key-name flag.
+
+If an application doesn't provide the --self-signing flag, it will be able to send relays
+to the AppGate server and it will still function as an application, however each request
+will have to contain the "?senderAddress=[address]" query parameter, where [address] is
+the address of the application that is sending the request. This is so that the server
+can generate the correct ring for the application and sign the request.`,
+		Args: cobra.NoArgs,
+		RunE: runAppGateServer,
+	}
+
+	cmd.Flags().StringVar(&signingKeyName, "signing-key-name", "", "The name of the key that will be used to sign relays")
+	cmd.Flags().StringVar(&listeningEndpoint, "listening-endpoint", "http://localhost:42069", "The host and port that the server will listen on")
+	cmd.Flags().StringVar(&cometWebsocketUrl, "comet-websocket-url", "ws://localhost:36657/websocket", "The URL of the tendermint websocket endpoint to interact with the chain")
+	cmd.Flags().BoolVar(&selfSigning, "self-signing", false, "Whether the server should sign all incoming requests with its own ring (for applications)")
+
+	cmd.Flags().String(flags.FlagKeyringBackend, "", "Select keyring's backend (os|file|kwallet|pass|test)")
+	cmd.Flags().String(flags.FlagNode, "tcp://localhost:36657", "tcp://<host>:<port> to tendermint rpc interface for this chain")
+
+	return cmd
+}
+
+func runAppGateServer(cmd *cobra.Command, _ []string) error {
+	// Create a context that is cancelled when the command is interrupted
+	ctx, cancelCtx := context.WithCancel(cmd.Context())
+
+	// Retrieve the client context for the chain interactions.
+	clientCtx := cosmosclient.GetClientContextFromCmd(cmd)
+
+	// Parse the listening endpoint.
+	listeningUrl, err := url.Parse(listeningEndpoint)
+	if err != nil {
+		cancelCtx()
+		return fmt.Errorf("failed to parse listening endpoint: %w", err)
+	}
+
+	// Obtain the tendermint websocket endpoint from the client context.
+	cometWSUrl, err := url.Parse(clientCtx.NodeURI + "/websocket")
+	if err != nil {
+		cancelCtx()
+		return fmt.Errorf("failed to parse block query URL: %w", err)
+	}
+	cometWSUrl.Scheme = "ws"
```

Comment:
Does `url.Parse` not populate this piece?

Reply:
I think I didn't understand what the default was and made sure it kept to websocket

Reply:
#PUC

Reply:
Removed entirely as we set a default 😅